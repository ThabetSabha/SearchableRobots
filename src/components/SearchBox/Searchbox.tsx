import React from "react";

const Searchbox = ({ searchChange }: SearchBoxProps) => {
  return (
    <>
      <input
        className="pa3 ma4 ba b--green bg-lightest-blue robots-search-input"
        type="search"
        placeholder="Search Robots"
        aria-label="search Robots"
        onChange={searchChange}
      ></input>
    </>
  );
};

export default Searchbox;
