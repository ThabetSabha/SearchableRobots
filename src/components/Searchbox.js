import React from 'react';


const Searchbox = ({ SearchChange }) => {
    return (
        <>
            <input
                className="pa3 ma4 ba b--green bg-lightest-blue"
                type="search"
                placeholder="Search Robots"
                onChange={SearchChange}
            ></input>
        </>
    )
}


export default Searchbox;