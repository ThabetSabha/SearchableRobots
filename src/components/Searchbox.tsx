import React from "react";

interface ISearchBoxProps {
  searchChange(event: React.SyntheticEvent<HTMLInputElement>): void;
}

const Searchbox = ({ searchChange }: ISearchBoxProps) => {
  return (
    <>
      <input
        className="pa3 ma4 ba b--green bg-lightest-blue"
        type="search"
        placeholder="Search Robots"
        aria-label="search Robots"
        onChange={searchChange}
      ></input>
    </>
  );
};

export default Searchbox;
