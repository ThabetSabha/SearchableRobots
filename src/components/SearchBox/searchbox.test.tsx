import React from "react";
import Searchbox from "./Searchbox";
import { shallow, ShallowWrapper } from "enzyme";

describe("<Searchbox/>", () => {
  let mockOnSearchChange: any;
  let searchboxProps: SearchBoxProps;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    mockOnSearchChange = jest.fn();
    searchboxProps = {
      searchChange: mockOnSearchChange,
    };

    wrapper = shallow(<Searchbox {...searchboxProps} />);
  });

  test("That the SearchBox Component Renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("that the onSearchChange function gets called on input change", () => {
    wrapper
      .find(".robots-search-input")
      .simulate("change", { target: { value: "test" } });
    expect(mockOnSearchChange).toHaveBeenCalledWith({
      target: { value: "test" },
    });
  });
});
