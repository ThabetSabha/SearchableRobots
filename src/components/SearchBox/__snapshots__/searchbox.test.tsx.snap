// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchbox/> That the SearchBox Component Renders 1`] = `
<Fragment>
  <input
    aria-label="search Robots"
    className="pa3 ma4 ba b--green bg-lightest-blue robots-search-input"
    onChange={[MockFunction]}
    placeholder="Search Robots"
    type="search"
  />
</Fragment>
`;
